internal/dag/gatewayapi_processor.go

```diff
@@ -61,6 +63,34 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 
 	for _, listener := range p.source.gateway.Spec.Listeners {
 
+		var listenerSecret *Secret
+		var err error
+
+		// Check for TLS on the Gateway.
+		if tls := listener.TLS; tls != nil {
+			// Validate the CertificateRef is configured.
+			if tls.CertificateRef == nil {
+				p.Error(contour_api_v1.ConditionTypeTLSError, "SecretNotConfigured",
+					"Spec.VirtualHost.TLS Secret is not configured")
```

Comment:
[field.Path](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation/field#Path) and [field.ErrorList](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation/field#ErrorList) is what I'm using upstream and provides helpful error context. [networking api validation](https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/apis/networking/validation/validation.go) is a good reference.